api/src/main/java/com/datastrato/gravitino/file/File.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.file;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.Namespace;
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * An interface representing a file in a schema {@link Namespace}. It defines the basic properties
+ * of a file object. A catalog implementation with {@link FileCatalog} should implement this
+ * interface.
+ */
+public interface File extends Auditable {
+
+  /** An enum representing the format of the file. */
+  enum Format {
+    CSV,
+    JSON,
+    PARQUET,
+    AVRO,
+    ORC,
+    TEXT,
+    BINARY,
+    IMAGE,
+    TFRECORD,
+    UNKNOWN
+  }
+
+  /** @return Name of the file object. */
+  String name();
```

Comment:
The `name` of the File entity is it's full path or just the name of it?

Reply:
Just the name. 

The virtual path is derived from the name identifier, and the physical path is a must-set property. About the physical path, I'm still thinking about using a property or a must-set field, I'm not so sure about which way is a good way.